src/pages/resultsView/oncoprint/AddClinicalTracks.tsx

```diff
@@ -0,0 +1,228 @@
+        await:()=>[this.props.store.clinicalAttributes, this.clinicalAttributeIdToAvailableFrequency],
+        invoke:()=>{
+            const availableFrequency = this.clinicalAttributeIdToAvailableFrequency.result!;
+            const sortedAttributes = {
+                clinical:[] as ClinicalAttribute[],
+                groups: [] as ClinicalAttribute[]
+            };
+            for (const attr of this.props.store.clinicalAttributes.result!) {
+                if (clinicalAttributeIsINCOMPARISONGROUP(attr)) {
+                    sortedAttributes.groups.push(attr);
+                } else {
+                    sortedAttributes.clinical.push(attr);
+                }
+            }
+            sortedAttributes.clinical = _.sortBy<ClinicalAttribute>(
+                sortedAttributes.clinical,
+                [   (x:ClinicalAttribute)=>{
+                    if (x.clinicalAttributeId === SpecialAttribute.StudyOfOrigin) {
+                        return 0;
+                    } else if (x.clinicalAttributeId === SpecialAttribute.MutationSpectrum) {
+                        return 1;
+                    } else if (clinicalAttributeIsPROFILEDIN(x)) {
+                        return 2;
+                    } else {
+                        return 3;
+                    }
+                },
+                    (x:ClinicalAttribute)=>{
+                        let freq = availableFrequency[x.clinicalAttributeId];
+                        if (freq === undefined) {
+                            freq = 0;
+                        }
+                        return -freq;
+                    },
+                    (x:ClinicalAttribute)=>-x.priority
+                    ,
+                    (x:ClinicalAttribute)=>x.displayName
+                ]
+            );
+
+            sortedAttributes.groups = _.sortBy<ClinicalAttribute>(sortedAttributes.groups, x=>x.displayName);
+
+            return Promise.resolve(_.mapValues(sortedAttributes, attrs=>{
+                return attrs.map(attr=>({
+                    label: attr.displayName,
+                    key: attr.clinicalAttributeId,
+                    selected: attr.clinicalAttributeId in this.selectedClinicalAttributeIds
+                }));
+            }));
+        }
+    });
+
+    readonly clinicalAttributeIdToAvailableFrequency = remoteData({
+        await:()=>[
+            this.props.store.clinicalAttributeIdToAvailableSampleCount,
+            this.props.store.samples
+        ],
+        invoke:()=>{
+            const numSamples = this.props.store.samples.result!.length;
+            return Promise.resolve(
+                _.mapValues(
+                    this.props.store.clinicalAttributeIdToAvailableSampleCount.result!,
+                    count=>100*count/numSamples
+                )
+            );
+        }
+    });
+
+    readonly addClinicalTracksMenu = MakeMobxView({
+        await:()=>[this.options],
+        render:()=>(
+            <AddChartByType
```

Comment:
should we generalize the name of this thing?

Reply:
yes probably though I don't know what it should be. maybe we should just leave it until we find a use that is different enough that it needs more generalizing